Snakefile

```diff
@@ -207,14 +208,68 @@ rule metrics_single:
         organism  = lambda wildcards: cfg.get_from_scenario(wildcards.scenario, key="organism"),
         assay     = lambda wildcards: cfg.get_from_scenario(wildcards.scenario, key="assay"),
         hvgs      = lambda wildcards: cfg.get_feature_selection(wildcards.hvg),
-        cmd       = f"conda run -n {cfg.py_env} python"	
+        cmd       = f"conda run -n {cfg.py_env} python"
     shell:
         """
         {params.cmd} {input.script} -u {input.u} -i {input.i} -o {output} -m {wildcards.method} \
         -b {params.batch_key} -l {params.label_key} --type {wildcards.o_type} \
         --hvgs {params.hvgs} --organism {params.organism} --assay {params.assay} -v
         """
 
+# ------------------------------------------------------------------------------
+# Save embeddings
+# ------------------------------------------------------------------------------
+
+rule embeddings_unintegrated:
+    input:
+        cfg.get_all_file_patterns("embeddings_unintegrated")
+    message:
+        "Collect all unintegrated embeddings"
+
+rule embeddings_integrated:
+    input: cfg.get_all_file_patterns("embeddings")
+    message: "Collect all integrated embeddings"
+
+all_embeddings = rules.embeddings_integrated.input
+if cfg.unintegrated_m:
+    all_embeddings.extend(rules.embeddings_unintegrated.input)
+
+rule embeddings:
+    input:
+        csvs = all_embeddings
+    output:
+        cfg.get_filename_pattern("embeddings", "final")
+    message:
+        "Completed all embeddings"
+    shell:
+     """
+     echo '{input.csvs}' | tr " " "\n" > {output}
+     """
+
+rule embeddings_single:
+    input:
+        adata  = get_integrated_for_metrics,
+        script = "scripts/save_embeddings.py"
```

Comment:
why is this an input? Isn't this always the same script? Could then move this to just be hardcoded in params or in shell directly.

Reply:
I usually add the script as input, which is convenient when we update the script. I.e. every time the timestamp of a script is changed, the step would be rerun.

With larger input, we might want to rethink this, as every update to the pipeline would force to rerun the pipeline, which we might not want. But then, we'll need to have a good overview which of the scripts are updated and need to be rerun.

Reply:
Typically, if we update a script we would be aware of what we're doing and therefore rerun the pipeline again as a result of that anyway I would imagine... otherwise why are you updating the script in the first place. Not sure you need an automated system taking care of the rerun there.

Reply:
I don't really have an opinion either way about what is more correct but my vote is leaving it this way for now for consistency with the other stages.